components/EventCards.vue

```diff
@@ -0,0 +1,199 @@
+import Guzel from './partials/Guzel.vue'
+import { useToast } from 'vue-toast-notification';
+
+
+interface Data {
+	date: string,
+	time: string,
+	place: string,
+	title: string,
+	description: string,
+	speaker: string,
+	address?: string
+}
+const props = defineProps<{
+	data: Data[]
+}>()
+
+const filterTab = ref<String>('')
+const results = computed(() => {
+	if (filterTab.value == 'Офлайн') {
+		return props.data.filter(item => item.place.toLowerCase() != 'онлайн')
+	}
+
+	if (filterTab.value == 'Онлайн') {
+		return props.data.filter(item => item.place.toLowerCase() == 'онлайн')
+	}
+	return props.data
+})
+
+
+const windowWidth = ref<number>(0)
+function getWindowWidth() {
+	windowWidth.value = window.innerWidth
+}
+
+let siteUrl = ref('')
+onMounted(() => {
+	getWindowWidth()
+	window.addEventListener('resize', getWindowWidth)
+	siteUrl.value = window.location.origin
+})
+
+const isExpanded = ref<boolean[]>([]);
+function toggleExpand(index: number) {
+	//to toggle expand for each separate card in a list
+	isExpanded.value[index] = isExpanded.value[index] ? !isExpanded.value[index] : true
+}
+
+function filterCards(tabName: string) {
+	filterTab.value = tabName
+}
+
+
+function getEventDate(item: Data) {
+	const regEx = /(\d{2})\.(\d{2})\.(\d{2})/
+	const parsedTime = item.time + ":00"
+	const parsedDate = item.date.replace(regEx, "20$3-$2-$1")
+	const eventTimeString = parsedDate + "T" + parsedTime + "+03:00"
+	return new Date(eventTimeString)
+}
+
+/* Days difference counts on time, so floating number is possible. 
+0 means that webinar takes place RIGHT NOW */
+const currentDate = computed(Date.now)
+function countDaysDifference(item: Data) {
+	const daysDifference = (getEventDate(item).getTime() - currentDate.value) / (1000 * 3600 * 24) //Transform ms to Days
+	return daysDifference
+}
+
+function hasEventPassed(item: Data) {
+	return countDaysDifference(item) < 0 ? true : false
+}
+
+function isWebinar(item: Data) {
+	return item.place.toLowerCase() == 'онлайн' ? true : false
+}
+
+const { data: landingDate, error } = await useFetch<string>(() =>
+	'https://academy.phcos.ru/api/date.json')
+
+function showCtaWebinar(item: Data) {
+	if ((new Date(landingDate.value!).getTime() == getEventDate(item).getTime())
+		&& !hasEventPassed(item)
+		&& isWebinar(item)) {
+		return true
+	} else {
+		return false
+	}
+}
+
+function showCtaOffline(item: Data) {
+	return !hasEventPassed(item) && !isWebinar(item) ? true : false
+}
+
+const $toast = useToast()
+const copyLink = async (item: Data) => {
+	const cardLink = siteUrl.value + '#event-' + item.date
+	try {
+		await navigator.clipboard.writeText(cardLink);
+		$toast.success('Cсылка скопирована!')
+	} catch (err) {
+		$toast.error('Ошибка копирования. Попробуйте еще раз')
+	}
+}
+</script>
+
+<template>
+	<Filters @tab-selected="filterCards" :places="props.data.map(item => item.place)" />
+	<TransitionGroup name="cards">
+		<section v-for="(item, index) in results"
```

Comment:
[Normal]
[Same comment as for `app.vue`](https://github.com/nikitakunevich/PhCosEvents-review/pull/1/files#r1638773502):
Creating subcomponents can help to improve readability.